apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/utils/TestUtils.java

```diff
@@ -0,0 +1,36 @@
+package utils;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+
+/** Utility class for tests */
+public class TestUtils {
+
+  private TestUtils() {
+    // Nothing to do.
+  }
+
+  /**
+   * Sets the value of a class field, even if it's private, and potentially if it's final
+   *
+   * @param object The object to change a field value for
+   * @param fieldName The name of the field to change the value for
+   * @param value The value to set the field to
+   * @throws IllegalAccessException If the field could not be accessed
+   * @throws NoSuchFieldException If the field did not exist on the class
+   */
+  public static void setField(Object object, String fieldName, Object value)
```

Comment:
Have you checked the build logs about this? Generally anything using reflection like this gets a warning to not do it and a notification that it will be deprecated in the future. (They've been trying to reduce the ability to do things like this for security reasons). Just trying to make sure we don't make a headache for ourselves later adding in hacks to modify the class for testing that will need to be changed anyway.

Reply:
This is for testing, without the ability to do this, it makes it harder to do proper testing with private/final fields without exposing it simply for testing reasons, which is arguably less secure than using reflection.  If the BFD team has guidance on this, we can follow what it recommends.

Reply:
Even for testing, it's still something that may be deprecated and need to be replaced later. Maybe monday we could discuss the methodology for testing things like this in the engineering sync to get some wider opinions.

Reply:
Trotter is out and many on the team will be out next Monday, so in the interest of not blocking this PR, let's call this fine for now and I'll put the broader discussion of reflection in testing for our next fully staffed Engineering Sync (probably 10/10 or 10/17)